sdk/react/panels/cast/CastDevicesScreen.js

```diff
@@ -1,45 +1,66 @@
-import React, { Component } from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import {
   Animated,
-  ListView,
   Text,
   View,
-  TouchableHighlight,
+  FlatList,
 } from 'react-native';
 
 import {
   BUTTON_NAMES,
 } from '../../constants';
+import CastDeviceListItem from './CastDeviceListItem';
 import Utils from '../../utils';
 import castDevicesStyles from '../style/CastDevicesStyles.json';
 
 const styles = Utils.getStyles(castDevicesStyles);
-
-let ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 === r2 });
-
 const dismissButtonSize = 20;
-const castButtonSize = 35;
 
-class CastDevicesScreen extends Component {
+export default class CastDevicesScreen extends PureComponent {
```

Comment:
Pure component rerenders only if props or state change, but it compares only **one level deep**, in other words it makes **shallow** comparison instead of deep. Please check that this component doesn't have **deep** props or state fields that can be changed during the component lifecycle.

Reply:
To simplify I suggest not to use `PureComponent` at this moment.

Reply:
@loginov-rocks  actually that's not true. `Component` rerenders every time its parent rerenders. `PureComponent` will do a shallow comparison before and it's a more preferable way to avoid unnecessary rerenders.
However, react-native had an issue when hot reload doesn't work for pure components. I'm not sure if it fixed or not

Reply:
done